Mathlib/Algebra/Periodic.lean

```diff
@@ -373,6 +374,11 @@ protected theorem Antiperiodic.funext' [Add α] [InvolutiveNeg β] (h : Antiperi
   neg_eq_iff_eq_neg.mpr h.funext
 #align function.antiperiodic.funext' Function.Antiperiodic.funext'
 
+/-- If a function is `antiperiodic` with antiperiod `c`, then it is also `periodic` with period
+  `2 • c`. -/
+protected theorem Antiperiodic.periodic_two_nsmul [AddMonoid α] [InvolutiveNeg β]
+    (h : Antiperiodic f c) : Periodic f (2 • c) := by simp [two_nsmul, ← add_assoc, h _]
+
 /-- If a function is `antiperiodic` with antiperiod `c`, then it is also `periodic` with period
   `2 * c`. -/
 protected theorem Antiperiodic.periodic [Semiring α] [InvolutiveNeg β] (h : Antiperiodic f c) :
```

Comment:
Can you replace this lemma with yours (keeping `Antiperiodic.periodic` as the name)?

Reply:
I've done the replacement. I kept the old `Antiperiodic.periodic` and renamed it to `Antiperiodic.periodic_two_mul`. I didn't want to delete it entirely because it aligns with the mathlib3 version.